### PR TITLE
fix: add more lint tests and fix flattening of requirements

### PR DESCRIPTION
**Pull request by wolfv**
There were a few places were requirements weren't properly flattened wrt to conditional statements in requirements. This fixes that.

### CODE REVIEW

**Comment by wolfv** [comment]
fix for https://github.com/conda-forge/conda-smithy/issues/2165

File: conda_smithy/linter/lints.py [hunk 1]
```diff
@@ -444,6 +451,7 @@ def lint_single_space_in_pinned_requirements(
 
             if symbol_to_check in req:
                 continue
+            print(req)
```
Comment (beckermr) [inline conda_smithy/linter/lints.py:454]:
```suggestion
```

**Review by beckermr — COMMENTED** [review]
One extra print.

**Comment by beckermr** [comment]
Also needs a news item.

File: conda_smithy/linter/lints.py (hunk 1 above)
Comment (wolfv) [reply conda_smithy/linter/lints.py]: Thanks!

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
Thanks @wolfv!

**Pull request merged**